File: lamb-upeu-sis/src/main/java/unmsm/edu/pe/tesis/domain/services/RevisorProyectoService.java
package unmsm.edu.pe.tesis.domain.services;

import unmsm.edu.pe.tesis.application.dto.ArchivoDescargable;
import unmsm.edu.pe.tesis.application.dto.EvaluacionRevisorResponse;
import unmsm.edu.pe.tesis.application.dto.EvaluarRevisorRequest;
import unmsm.edu.pe.tesis.application.dto.ObservarItemRequest;
import unmsm.edu.pe.tesis.application.dto.RevisorBandejaItem;

import java.util.List;
import java.util.UUID;

/** Casos de uso del REVISOR (Jurado Informante) sobre el proyecto (Etapa 5, paso 3). */
public interface RevisorProyectoService {

    /** Proyectos asignados al docente revisor actual. */
    List<RevisorBandejaItem> bandeja();

    /** Detalle del proyecto (solo lectura) + rúbrica con la evaluación del revisor. */
    EvaluacionRevisorResponse detalle(UUID tesisId);

    /** Descarga la rúbrica oficial (Word) que subió Secretaría, solo para referencia del revisor. */
    ArchivoDescargable descargarRubrica(UUID tesisId);

    /** Descarga la rúbrica ya llenada (Word) con los niveles y puntajes que el revisor registró. */
    ArchivoDescargable descargarRubricaLlenada(UUID tesisId);

    /** Genera el PDF del proyecto para el visor del revisor (solo lectura). */
    ArchivoDescargable descargarProyectoPdf(UUID tesisId);

    /** Observa un ítem/campo del proyecto (queda en el historial del estudiante como informe de revisor). */
    void observarItem(UUID tesisId, ObservarItemRequest req);

    /** Valida la corrección de un ítem: lo da por conforme (el estudiante ya no debe corregirlo). */
    void darConformidadItem(UUID tesisId, String campo);

    /** Registra la evaluación con rúbrica (observa o da conformidad). */
    void evaluar(UUID tesisId, EvaluarRevisorRequest req);
}

File: themeforest-Fj3VzBCp-fuse-angularjs-material-design-admin-template/fuse/src/app/views/dashboard/revisor-proyecto/services/revisor-proyecto.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '@/environments/environment';
import { END_POINTS } from '@/app/providers/utils/end-points';

@Injectable({ providedIn: 'root' })
export class RevisorProyectoService {
  private _http = inject(HttpClient);
  private readonly base = environment.url + END_POINTS.revisor.proyectos;

  bandeja$(): Observable<any> {
    return this._http.get<any>(this.base);
  }

  detalle$(tesisId: string): Observable<any> {
    return this._http.get<any>(`${this.base}/${tesisId}`);
  }

  evaluar$(tesisId: string, niveles: Record<string, string>, observaciones: Record<string, string>, comentario: string, conforme: boolean): Observable<any> {
    return this._http.post<any>(`${this.base}/${tesisId}/evaluar`, { niveles, observaciones, comentario, conforme });
  }

  /** PDF del proyecto para el visor (blob con auth). */
  proyectoPdf$(tesisId: string): Observable<Blob> {
    return this._http.get(`${this.base}/${tesisId}/proyecto-pdf`, { responseType: 'blob' });
  }

  /** Descarga la rúbrica oficial (Word) subida por Secretaría — con auth (blob). */
  descargarRubrica$(tesisId: string): Observable<Blob> {
    return this._http.get(`${this.base}/${tesisId}/rubrica`, { responseType: 'blob' });
  }

  /** Descarga la rúbrica ya llenada (Excel) con los puntajes del revisor. */
  descargarRubricaLlenada$(tesisId: string): Observable<Blob> {
    return this._http.get(`${this.base}/${tesisId}/rubrica-llenada`, { responseType: 'blob' });
  }

  observarItem$(tesisId: string, campo: string, texto: string): Observable<any> {
    return this._http.post<any>(`${this.base}/${tesisId}/observar-item`, { campo, texto });
  }

  conformidadItem$(tesisId: string, campo: string): Observable<any> {
    return this._http.post<any>(`${this.base}/${tesisId}/items/${campo}/conformidad`, {});
  }
}
